util/network-devp2p/src/node_table.rs

```diff
@@ -111,6 +111,14 @@ impl NodeEndpoint {
 		self.to_rlp(rlp);
 	}
 
+	pub fn to_enr(&self, key: &parity_crypto::publickey::Secret) -> enr::Enr {
+		let mut builder = enr::EnrBuilder::new("v4");
+		builder.ip(self.address.ip());
+		builder.tcp(self.address.port());
+		builder.udp(self.udp_port);
+		builder.build(&enr::DefaultKey::Secp256k1(secp256k1::SecretKey::parse_slice((*key).as_bytes()).expect("rust-secp256k1 keychain is always valid; qed"))).expect("ENR always correct; qed")
```

Comment:
Why is `ENR always correct`? While the first short proof makes sense to me, as we are converting between representations from different secp256k1 libraries, the second one should be more elaborate on why we satisfy all the conditions for `build`.
Like right now I don't even know how is it ever possible for `build()` to fail, so there is no way to verify that we actually do it here.